spark/src/main/scala/org/apache/spark/sql/delta/actions/TableFeatureSupport.scala

```diff
@@ -498,4 +498,12 @@ object TableFeatureProtocolUtils {
    */
   def minimumRequiredVersions(features: Seq[TableFeature]): (Int, Int) =
     ((features.map(_.minReaderVersion) :+ 1).max, (features.map(_.minWriterVersion) :+ 1).max)
+
+  /**
+   * Return a set with the implicit features of the provided feature set.
+   */
+  def implicitFeatures(features: Set[TableFeature]): Set[TableFeature] =
```

Comment:
I would like to name this method more clearly

```suggestion
  /**
   * Return a set of legacy features that will be supported by a legacy protocol
   * as a "byproduct" of supporting the given legacy `features`.
   *
   * As an example, the legacy protocol for supporting columnMapping also supports
   * x, x, and x as byproducts, and there is no way to not support them.
   */
  def implicitLegacyFeatures(features: Set[TableFeature]): Set[TableFeature] =
```